executioncontext/workspacecontext/src/main/java/org/jetbrains/bsp/bazel/workspacecontext/TargetsSpec.kt

```diff
@@ -23,24 +25,62 @@ internal object TargetsSpecMapper : ProjectViewToExecutionContextEntityMapper<Ta
+
+    private fun deriveTargetsFlagFalse(projectView: ProjectView): Try<TargetsSpec> =
         when {
             projectView.targets == null -> Try.success(defaultTargetsSpec)
             hasEmptyIncludedValuesAndEmptyExcludedValues(projectView.targets!!) -> Try.success(defaultTargetsSpec)
             hasEmptyIncludedValuesAndNonEmptyExcludedValues(projectView.targets!!) -> Try.failure(
-                ProjectViewToExecutionContextEntityMapperException(
-                    NAME, "'targets' section has no included targets."
-                )
+                    ProjectViewToExecutionContextEntityMapperException(
+                            NAME, "'targets' section has no included targets."
+                    )
             )
-            else -> Try.success(mapNotEmptySection(projectView.targets!!))
+            else -> Try.success(mapNotEmptyNotDerivedTargetsSection(projectView.targets!!))
         }
 
     private fun hasEmptyIncludedValuesAndEmptyExcludedValues(targetsSection: ProjectViewTargetsSection): Boolean =
         targetsSection.values.isEmpty() and targetsSection.excludedValues.isEmpty()
     private fun hasEmptyIncludedValuesAndNonEmptyExcludedValues(targetsSection: ProjectViewTargetsSection): Boolean =
         targetsSection.values.isEmpty() and targetsSection.excludedValues.isNotEmpty()
 
-    private fun mapNotEmptySection(targetsSection: ProjectViewTargetsSection): TargetsSpec =
+    private fun hasEmptyIncludedValuesAndEmptyExcludedValuesDirectories(directoriesSection: ProjectViewDirectoriesSection): Boolean =
+        directoriesSection.values.isEmpty() and directoriesSection.excludedValues.isEmpty()
+    private fun hasEmptyIncludedValuesAndNonEmptyExcludedValuesDirectories(directoriesSection: ProjectViewDirectoriesSection): Boolean =
+        directoriesSection.values.isEmpty() and directoriesSection.excludedValues.isNotEmpty()
+
+    private fun deriveTargetsFlags(deriveTargetsFlagSection: ProjectViewDeriveTargetsFlagSection): Boolean =
+        deriveTargetsFlagSection.value
+
+    private fun mapNotEmptyNotDerivedTargetsSection(targetsSection: ProjectViewTargetsSection): TargetsSpec =
         TargetsSpec(targetsSection.values, targetsSection.excludedValues)
 
+    private fun mapNotEmptyDerivedTargetSection(targetsSection: ProjectViewTargetsSection?, directoriesSection: ProjectViewDirectoriesSection): TargetsSpec {
+        val directoriesValues = directoriesSection.values.map { mapDirectoryToTarget(it) }
+        val directoriesExcludedValues = directoriesSection.excludedValues.map { mapDirectoryToTarget(it) }
+
+        return TargetsSpec(
+                (targetsSection?.values ?: emptyList()) + directoriesValues,
+                (targetsSection?.excludedValues ?: emptyList()) + directoriesExcludedValues)
+    }
+
+    private fun mapDirectoryToTarget(buildDirectoryIdentifier: BuildTargetIdentifier): BuildTargetIdentifier =
+            BuildTargetIdentifier("//" + buildDirectoryIdentifier.uri + "/...")
```

Comment:
it'll fail if someone will specify directory as `dir/` -> `//dir//...` (tests needed)

Reply:
Added check for this special case